frontend/src/app/highlight/highlight.component.ts

```diff
@@ -9,13 +11,9 @@ import {AlertService} from '../alert.service';
   styleUrls: ['./highlight.component.css']
 })
 export class HighlightComponent implements OnInit {
-  alert: Alert;
+  alert$: Observable<Alert> = this.alertService.getMessage().pipe(shareReplay());
```

Comment:
`shareReplay` is not trivial to understand without good understanding of rxjs. Could the same purpose be achieved without the shareReplay call? That role this it play here? Also can you comment on the dollar sign at the end? Is it rxjs standard code style?

Reply:
Could the same purpose be achieved without the shareReplay call?
  1. We can leave it as it was, just add **takeUntil** and implement **OnDestroy** (I used that feature as an example to automatic unsubscribe observables using the **async** directive in the template).

That role this it play here?
  2. shareReplay is used there as a safeguard for example, let suppose **alertService.getMessage()** is used by multiple components, it will throw duplicate HTTP requests. With shareReplay() we can avoid that behavior.
  
Also can you comment on the dollar sign at the end?
  Observable stream code style